test/hotspot/jtreg/runtime/ParallelLoad/SuperWait/SuperWaitTest.java

```diff
@@ -36,19 +36,17 @@
 
 public class SuperWaitTest {
 
-    private static Semaphore mainSync = null;
+    private static volatile boolean dIsLoading = false;
```

Comment:
This doesn't need to be volatile as it is only accessed under the classloader lock.

Reply:
ThreadPrint.println("Waiting for " + name);
                    while (!dIsLoading) {  // guard against spurious wakeup
                        try {
                            wait(); // let the other thread have this lock.
                        } catch (InterruptedException ie) {}
                    }

Do we not have to make dIsLoading volatile?